services/blockstorage/sync/state_finished_car_test.go

```diff
@@ -0,0 +1,54 @@
+package sync
+
+import (
+	"github.com/orbs-network/orbs-network-go/test/builders"
+	"github.com/orbs-network/orbs-spec/types/go/protocol/gossipmessages"
+	"github.com/stretchr/testify/require"
+	"testing"
+)
+
+func TestFinishedWithNoResponsesGoBackToIdle(t *testing.T) {
+	h := newBlockSyncHarness()
+	finishedState := h.sf.CreateFinishedCARState([]*gossipmessages.BlockAvailabilityResponseMessage{})
+	shouldBeIdleState := finishedState.processState(h.ctx)
+
+	require.IsType(t, &idleState{}, shouldBeIdleState, "next state should be idle")
+}
+
+func TestFinishedWithResponsesMoveToWaitingForChunk(t *testing.T) {
+	response := builders.BlockAvailabilityResponseInput().Build().Message
+	h := newBlockSyncHarness()
+	finishedState := h.sf.CreateFinishedCARState([]*gossipmessages.BlockAvailabilityResponseMessage{response})
+	shouldBeWaitingState := finishedState.processState(h.ctx)
+
+	require.IsType(t, &waitingForChunksState{}, shouldBeWaitingState, "next state should be waiting for chunk")
+}
+
+func TestFinishedWithInvalidResponsesMovesToIdle(t *testing.T) {
+	h := newBlockSyncHarness()
+	finishedState := h.sf.CreateFinishedCARState(nil)
```

Comment:
again, I think nil slices and empty slices are the same, am I wrong?

Reply:
yes, so can it ever be "invalid"? i think i should leave it as sanity

Reply:
I think not. It's a property of golang that slices are never nil